src/codeu/chat/server/Model.java

```diff
@@ -68,12 +78,42 @@ public int compare(Time a, Time b) {
   private final Uuid.Generator userGenerations = new LinearUuidGenerator(null, 1, Integer.MAX_VALUE);
   private Uuid currentUserGeneration = userGenerations.make();
 
+  private final DatabaseAccess access = new DatabaseAccess();
+  private DatabaseReference ref = access.initialize();
+
+  // users
+  public void loadUsers() {
+    DatabaseReference usersRef = ref.child("users");
+    usersRef.addListenerForSingleValueEvent(new ValueEventListener() {
+      @Override
+      public void onDataChange(DataSnapshot dataSnapshot) {
+        System.out.println("Users Loaded: " + dataSnapshot.toString());
+        // TODO: fix
+        for (DataSnapshot child : dataSnapshot.getChildren()) {
+          UserModel user = child.getValue(UserModel.class);
+          System.out.println("Name: " + user.name + ", Creation: " + user.creation);
+        }
+      }
+
+      @Override
+      public void onCancelled(DatabaseError databaseError) {
+
+      }
+    });
+  }
+
   public void add(User user) {
     currentUserGeneration = userGenerations.make();
+    DatabaseReference usersRef = ref.child("users");
 
     userById.insert(user.id, user);
```

Comment:
If we're using firebase as backend I think we can just delete the google provided data store code